contracts/SnappAuction.sol

```diff
@@ -68,13 +70,39 @@ contract SnappAuction is SnappBase {
         auctions[auctionIndex].size++;
     }
 
+    function applyAuction(
+        uint slot,
+        bytes32 _currStateRoot,
+        bytes32 _newStateRoot,
+        bytes32 _orderHash,
+        uint128[MAX_TOKENS] memory prices,
+        uint128[2*AUCTION_BATCH_SIZE] memory volumes
+    )
+        public onlyOwner()
+    {   
+        require(slot <= auctionIndex, "Requested order slot does not exist");
+        require(slot == 0 || auctions[slot-1].appliedAccountStateIndex != 0, "Must apply auction slots in order!");
+        require(auctions[slot].shaHash == _orderHash, "Order hash doesn't agree");
+        require(auctions[slot].appliedAccountStateIndex == 0, "Auction already applied");
+        require(block.number > auctions[slot].creationBlock + 20, "Requested order slot is still active");
```

Comment:
Could the block already be inactive if it was full?

Reply:
Not according to the definition of active provided here. This imposes a hard limit on the notion of an active slot.

Reply:
Is there a technical reason why we shouldn't allow to process a batch as soon as it's been closed? If I understand correctly a batch is closed as soon as it's full and not necessarily after three minutes.

Reply:
This is true, we could easily implement this change (although I kind of assumed we would more often reach batch-time before batch capacity). I will add this to the list of todos. Does this imply we are going to do the same for deposits and withdraws?

Reply:
Done!

Reply:
Done!